modules/seqexec/server/src/test/scala/seqexec/server/tcs/TestTcsEpics.scala

```diff
@@ -0,0 +1,1067 @@
+package seqexec.server.tcs
+import cats.{Applicative, Eq}
+import cats.effect.{Sync, Timer}
+import cats.effect.concurrent.Ref
+import cats.implicits._
+import edu.gemini.seqexec.server.tcs.{BinaryOnOff, BinaryYesNo}
+import monocle.{Getter, Lens}
+import monocle.macros.Lenses
+import seqexec.model.enum.ApplyCommandResult
+import seqexec.server.TestEpicsCommand._
+import seqexec.server.EpicsCommand
+import seqexec.server.tcs.TcsEpics._
+import squants.{Angle, Time}
+import squants.space.AngleConversions._
+
+class TestTcsEpics[F[_]: Sync](state: Ref[F, TestTcsEpics.State], out: Ref[F, List[TestTcsEpics.TestTcsEvent]])
+  extends TcsEpics[F] {
+  import TestTcsEpics._
+
+  val outputF: F[List[TestTcsEvent]] = out.get
+
+  override def post: F[ApplyCommandResult] =
+    List[EpicsCommand[F]](
+      m1GuideCmd,
+      m2GuideCmd,
+      m2GuideModeCmd,
+      m2GuideModeCmd,
+      m2GuideConfigCmd,
+      mountGuideCmd,
+      offsetACmd,
+      offsetBCmd,
+      pwfs1Park,
+      pwfs1ProbeFollowCmd,
+      pwfs1ProbeGuideCmd,
+      pwfs1ObserveCmd,
+      pwfs1StopObserveCmd,
+      pwfs2Park,
+      pwfs2ProbeFollowCmd,
+      pwfs2ProbeGuideCmd,
+      pwfs2ObserveCmd,
+      pwfs2StopObserveCmd,
+      oiwfsPark,
+      oiwfsProbeFollowCmd,
+      oiwfsProbeGuideCmd,
+      oiwfsObserveCmd,
+      oiwfsStopObserveCmd,
+      offsetACmd
+    ).map(_.post)
+      .sequence
+      .map(_.find(_ =!= ApplyCommandResult.Completed).getOrElse(ApplyCommandResult.Completed))
```

Comment:
Returns the first error, or completed otherwise. Right now no commands returns an error, but it could be useful in the future.